compiler/src/iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -278,6 +278,10 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
     buildHALConfigurationPassPipeline(passManager, targetRegistry,
                                       targetOptions, hooks);
 
+    // HACK: this should not be here and will be going away. It exists for
+    // lowering iree_linalg_ext.upper_bound_tile_size ops that exist on the
+    // host. We should be using stream ops for performing such calculations that
+    // we can attach affinities to and understand what devices are being used.
     FunctionLikeNest(passManager).addPass([]() {
       return createCPUMaterializeUpperBoundTileSizePass();
     });
```

Comment:
Is this cleanup tracked somewhere?

Reply:
I think the data tiling work is tracking it - chatted with hanhan today and he said it's going away soon(ish)! woohoo!

Reply:
Yes, it's coming! @Max191 is working on it. The 1/n draft PR is https://github.com/iree-org/iree/pull/17931 Then we will be able to remove the op and the pass entirely.